internal/updater/updater.go

```diff
@@ -230,6 +232,19 @@ func (u *UpdateInstaller) InstallBlocking(installTargetPath string, args ...stri
 		return errs.Wrap(err, "Could not run installer")
 	}
 
+	// installerPath looks like "<tempDir>/state-update\d{10}/state-install/state-installer".
+	updateDir := filepath.Dir(filepath.Dir(installerPath))
+	logging.Debug("Cleaning up temporary update directory: %s", updateDir)
+	if strings.HasPrefix(filepath.Base(updateDir), "state-update") {
+		err = os.RemoveAll(updateDir)
+		if err != nil {
+			multilog.Error("Unable to remove update directory '%s': %v", updateDir, err)
+		}
+	} else {
+		multilog.Error("Update directory structure has changed without updating it here, too. "+
+			"installerPath: %s\nupdateDir: %s\nExpected a 'state-update' prefix for the latter", installerPath, updateDir)
```

Comment:
I can see us changing the structure of updates and getting a bunch of these errors on Rollbar from older versions. I think the error above is enough to report.